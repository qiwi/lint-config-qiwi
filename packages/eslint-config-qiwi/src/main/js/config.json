{
  "env": {
    "es2022": true,
    "browser": true,
    "node": true,
    "jest": true
  },
  "settings": {
    "react": {
      "version": "18"
    }
  },
  "plugins": [
    "react",
    "react-hooks",
    "sonarjs",
    "unicorn"
  ],
  "parser": "espree",
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true
    },
    "jsxPragma": null,
    "ecmaVersion": "latest",
    "sourceType": "module"
  },
  "rules": {
    "array-bracket-newline": "off",
    "array-bracket-spacing": "off",
    "array-element-newline": "off",
    "arrow-parens": "off",
    "arrow-spacing": "off",
    "block-spacing": "off",
    "brace-style": "off",
    "comma-dangle": "off",
    "comma-spacing": "off",
    "comma-style": "off",
    "computed-property-spacing": "off",
    "constructor-super": "error",
    "curly": "off",
    "dot-location": "off",
    "eol-last": "off",
    "for-direction": "error",
    "func-call-spacing": "off",
    "function-call-argument-newline": "off",
    "function-paren-newline": "off",
    "generator-star": "off",
    "generator-star-spacing": "off",
    "getter-return": "error",
    "implicit-arrow-linebreak": "off",
    "indent": "off",
    "indent-legacy": "off",
    "jsx-quotes": "off",
    "key-spacing": "off",
    "keyword-spacing": "off",
    "linebreak-style": "off",
    "lines-around-comment": "off",
    "max-len": "off",
    "multiline-ternary": "off",
    "new-parens": "off",
    "newline-per-chained-call": "off",
    "no-array-constructor": "off",
    "no-arrow-condition": "off",
    "no-async-promise-executor": "error",
    "no-case-declarations": "error",
    "no-class-assign": "error",
    "no-comma-dangle": "off",
    "no-compare-neg-zero": "error",
    "no-cond-assign": "error",
    "no-confusing-arrow": "off",
    "no-const-assign": "error",
    "no-constant-condition": "error",
    "no-control-regex": "error",
    "no-debugger": "error",
    "no-delete-var": "error",
    "no-dupe-args": "error",
    "no-dupe-class-members": "error",
    "no-dupe-else-if": "error",
    "no-dupe-keys": "error",
    "no-duplicate-case": "error",
    "no-empty": "error",
    "no-empty-character-class": "error",
    "no-empty-function": "off",
    "no-empty-pattern": "error",
    "no-ex-assign": "error",
    "no-extra-boolean-cast": "error",
    "no-extra-parens": "off",
    "no-extra-semi": "off",
    "no-fallthrough": "error",
    "no-floating-decimal": "off",
    "no-func-assign": "error",
    "no-global-assign": "error",
    "no-import-assign": "error",
    "no-inner-declarations": "error",
    "no-invalid-regexp": "error",
    "no-irregular-whitespace": "error",
    "no-loss-of-precision": "off",
    "no-misleading-character-class": "error",
    "no-mixed-operators": "off",
    "no-mixed-spaces-and-tabs": "off",
    "no-multi-spaces": "off",
    "no-multiple-empty-lines": "off",
    "no-negated-condition": "off",
    "no-nested-ternary": "off",
    "no-new-symbol": "error",
    "no-nonoctal-decimal-escape": "error",
    "no-obj-calls": "error",
    "no-octal": "error",
    "no-prototype-builtins": "error",
    "no-redeclare": "error",
    "no-regex-spaces": "error",
    "no-reserved-keys": "off",
    "no-self-assign": "error",
    "no-setter-return": "error",
    "no-shadow-restricted-names": "error",
    "no-space-before-semi": "off",
    "no-spaced-func": "off",
    "no-sparse-arrays": "error",
    "no-tabs": "off",
    "no-this-before-super": "error",
    "no-trailing-spaces": "off",
    "no-undef": "error",
    "no-unexpected-multiline": "off",
    "no-unreachable": "error",
    "no-unsafe-finally": "error",
    "no-unsafe-negation": "error",
    "no-unsafe-optional-chaining": "error",
    "no-unused-labels": "error",
    "no-unused-vars": "off",
    "no-useless-backreference": "error",
    "no-useless-catch": "error",
    "no-useless-escape": "error",
    "no-whitespace-before-property": "off",
    "no-with": "error",
    "no-wrap-func": "off",
    "nonblock-statement-body-position": "off",
    "object-curly-newline": "off",
    "object-curly-spacing": "off",
    "object-property-newline": "off",
    "one-var-declaration-per-line": "off",
    "operator-linebreak": "off",
    "padded-blocks": "off",
    "quote-props": "off",
    "quotes": "off",
    "require-yield": "error",
    "rest-spread-spacing": "off",
    "semi": "off",
    "semi-spacing": "off",
    "semi-style": "off",
    "space-after-function-name": "off",
    "space-after-keywords": "off",
    "space-before-blocks": "off",
    "space-before-function-paren": "off",
    "space-before-function-parentheses": "off",
    "space-before-keywords": "off",
    "space-in-brackets": "off",
    "space-in-parens": "off",
    "space-infix-ops": "off",
    "space-return-throw-case": "off",
    "space-unary-ops": "off",
    "space-unary-word-ops": "off",
    "switch-colon-spacing": "off",
    "template-curly-spacing": "off",
    "template-tag-spacing": "off",
    "unicode-bom": "off",
    "use-isnan": "error",
    "valid-typeof": "error",
    "wrap-iife": "off",
    "wrap-regex": "off",
    "yield-star-spacing": "off",
    "@babel/object-curly-spacing": "off",
    "@babel/semi": "off",
    "babel/object-curly-spacing": "off",
    "babel/quotes": "off",
    "babel/semi": "off",
    "flowtype/boolean-style": "off",
    "flowtype/delimiter-dangle": "off",
    "flowtype/generic-spacing": "off",
    "flowtype/object-type-curly-spacing": "off",
    "flowtype/object-type-delimiter": "off",
    "flowtype/quotes": "off",
    "flowtype/semi": "off",
    "flowtype/space-after-type-colon": "off",
    "flowtype/space-before-generic-bracket": "off",
    "flowtype/space-before-type-colon": "off",
    "flowtype/union-intersection-spacing": "off",
    "react-hooks/exhaustive-deps": "warn",
    "react-hooks/rules-of-hooks": "error",
    "react/display-name": "error",
    "react/jsx-child-element-spacing": "off",
    "react/jsx-closing-bracket-location": "off",
    "react/jsx-closing-tag-location": "off",
    "react/jsx-curly-newline": "off",
    "react/jsx-curly-spacing": "off",
    "react/jsx-equals-spacing": "off",
    "react/jsx-first-prop-new-line": "off",
    "react/jsx-indent": "off",
    "react/jsx-indent-props": "off",
    "react/jsx-key": "error",
    "react/jsx-max-props-per-line": "off",
    "react/jsx-newline": "off",
    "react/jsx-no-comment-textnodes": "error",
    "react/jsx-no-duplicate-props": "error",
    "react/jsx-no-target-blank": "error",
    "react/jsx-no-undef": "error",
    "react/jsx-one-expression-per-line": "off",
    "react/jsx-props-no-multi-spaces": "off",
    "react/jsx-space-before-closing": "off",
    "react/jsx-tag-spacing": "off",
    "react/jsx-uses-react": "off",
    "react/jsx-uses-vars": "error",
    "react/jsx-wrap-multilines": "off",
    "react/no-children-prop": "error",
    "react/no-danger-with-children": "error",
    "react/no-deprecated": "error",
    "react/no-direct-mutation-state": "error",
    "react/no-find-dom-node": "error",
    "react/no-is-mounted": "error",
    "react/no-render-return-value": "error",
    "react/no-string-refs": "error",
    "react/no-unescaped-entities": "error",
    "react/no-unknown-property": "error",
    "react/no-unsafe": "off",
    "react/prop-types": "error",
    "react/react-in-jsx-scope": "off",
    "react/require-render-return": "error",
    "sonarjs/cognitive-complexity": "error",
    "sonarjs/elseif-without-else": "off",
    "sonarjs/max-switch-cases": "error",
    "sonarjs/no-all-duplicated-branches": "error",
    "sonarjs/no-collapsible-if": "error",
    "sonarjs/no-collection-size-mischeck": "error",
    "sonarjs/no-duplicate-string": "error",
    "sonarjs/no-duplicated-branches": "error",
    "sonarjs/no-element-overwrite": "error",
    "sonarjs/no-empty-collection": "error",
    "sonarjs/no-extra-arguments": "error",
    "sonarjs/no-gratuitous-expressions": "error",
    "sonarjs/no-identical-conditions": "error",
    "sonarjs/no-identical-expressions": "error",
    "sonarjs/no-identical-functions": "error",
    "sonarjs/no-ignored-return": "error",
    "sonarjs/no-inverted-boolean-check": "off",
    "sonarjs/no-nested-switch": "error",
    "sonarjs/no-nested-template-literals": "error",
    "sonarjs/no-one-iteration-loop": "error",
    "sonarjs/no-redundant-boolean": "error",
    "sonarjs/no-redundant-jump": "error",
    "sonarjs/no-same-line-conditional": "error",
    "sonarjs/no-small-switch": "error",
    "sonarjs/no-unused-collection": "error",
    "sonarjs/no-use-of-empty-return-value": "error",
    "sonarjs/no-useless-catch": "error",
    "sonarjs/non-existent-operator": "error",
    "sonarjs/prefer-immediate-return": "error",
    "sonarjs/prefer-object-literal": "error",
    "sonarjs/prefer-single-boolean-return": "error",
    "sonarjs/prefer-while": "error",
    "standard/array-bracket-even-spacing": "off",
    "standard/computed-property-even-spacing": "off",
    "standard/object-curly-even-spacing": "off",
    "unicorn/better-regex": "error",
    "unicorn/catch-error-name": "off",
    "unicorn/consistent-destructuring": "error",
    "unicorn/consistent-function-scoping": "error",
    "unicorn/custom-error-definition": "off",
    "unicorn/empty-brace-spaces": "off",
    "unicorn/error-message": "error",
    "unicorn/escape-case": "error",
    "unicorn/expiring-todo-comments": "error",
    "unicorn/explicit-length-check": "error",
    "unicorn/filename-case": "off",
    "unicorn/import-style": "error",
    "unicorn/new-for-builtins": "error",
    "unicorn/no-abusive-eslint-disable": "off",
    "unicorn/no-array-callback-reference": "off",
    "unicorn/no-array-for-each": "off",
    "unicorn/no-array-method-this-argument": "error",
    "unicorn/no-array-push-push": "error",
    "unicorn/no-array-reduce": "off",
    "unicorn/no-await-expression-member": "warn",
    "unicorn/no-console-spaces": "error",
    "unicorn/no-document-cookie": "error",
    "unicorn/no-empty-file": "error",
    "unicorn/no-for-loop": "error",
    "unicorn/no-hex-escape": "error",
    "unicorn/no-instanceof-array": "error",
    "unicorn/no-invalid-remove-event-listener": "error",
    "unicorn/no-keyword-prefix": "off",
    "unicorn/no-lonely-if": "error",
    "unicorn/no-negated-condition": "error",
    "unicorn/no-nested-ternary": "off",
    "unicorn/no-new-array": "error",
    "unicorn/no-new-buffer": "error",
    "unicorn/no-null": "off",
    "unicorn/no-object-as-default-parameter": "error",
    "unicorn/no-process-exit": "error",
    "unicorn/no-static-only-class": "error",
    "unicorn/no-thenable": "error",
    "unicorn/no-this-assignment": "error",
    "unicorn/no-typeof-undefined": "error",
    "unicorn/no-unnecessary-await": "error",
    "unicorn/no-unreadable-array-destructuring": "error",
    "unicorn/no-unreadable-iife": "error",
    "unicorn/no-unsafe-regex": "off",
    "unicorn/no-unused-properties": "off",
    "unicorn/no-useless-fallback-in-spread": "error",
    "unicorn/no-useless-length-check": "error",
    "unicorn/no-useless-promise-resolve-reject": "error",
    "unicorn/no-useless-spread": "error",
    "unicorn/no-useless-switch-case": "error",
    "unicorn/no-useless-undefined": "error",
    "unicorn/no-zero-fractions": "error",
    "unicorn/number-literal-case": "off",
    "unicorn/numeric-separators-style": "error",
    "unicorn/prefer-add-event-listener": "error",
    "unicorn/prefer-array-find": "error",
    "unicorn/prefer-array-flat": "error",
    "unicorn/prefer-array-flat-map": "error",
    "unicorn/prefer-array-index-of": "error",
    "unicorn/prefer-array-some": "error",
    "unicorn/prefer-at": "off",
    "unicorn/prefer-code-point": "error",
    "unicorn/prefer-date-now": "error",
    "unicorn/prefer-default-parameters": "error",
    "unicorn/prefer-dom-node-append": "error",
    "unicorn/prefer-dom-node-dataset": "error",
    "unicorn/prefer-dom-node-remove": "error",
    "unicorn/prefer-dom-node-text-content": "error",
    "unicorn/prefer-event-target": "off",
    "unicorn/prefer-export-from": "error",
    "unicorn/prefer-includes": "error",
    "unicorn/prefer-json-parse-buffer": "off",
    "unicorn/prefer-keyboard-event-key": "error",
    "unicorn/prefer-logical-operator-over-ternary": "error",
    "unicorn/prefer-math-trunc": "off",
    "unicorn/prefer-modern-dom-apis": "error",
    "unicorn/prefer-modern-math-apis": "error",
    "unicorn/prefer-module": "off",
    "unicorn/prefer-native-coercion-functions": "error",
    "unicorn/prefer-negative-index": "error",
    "unicorn/prefer-node-protocol": "error",
    "unicorn/prefer-number-properties": "error",
    "unicorn/prefer-object-from-entries": "error",
    "unicorn/prefer-optional-catch-binding": "error",
    "unicorn/prefer-prototype-methods": "error",
    "unicorn/prefer-query-selector": "error",
    "unicorn/prefer-reflect-apply": "error",
    "unicorn/prefer-regexp-test": "error",
    "unicorn/prefer-set-has": "error",
    "unicorn/prefer-set-size": "error",
    "unicorn/prefer-spread": "error",
    "unicorn/prefer-string-replace-all": "off",
    "unicorn/prefer-string-slice": "error",
    "unicorn/prefer-string-starts-ends-with": "error",
    "unicorn/prefer-string-trim-start-end": "error",
    "unicorn/prefer-switch": "error",
    "unicorn/prefer-ternary": "error",
    "unicorn/prefer-top-level-await": "warn",
    "unicorn/prefer-type-error": "error",
    "unicorn/prevent-abbreviations": "off",
    "unicorn/relative-url-style": "error",
    "unicorn/require-array-join-separator": "error",
    "unicorn/require-number-to-fixed-digits-argument": "error",
    "unicorn/require-post-message-target-origin": "off",
    "unicorn/string-content": "off",
    "unicorn/switch-case-braces": "off",
    "unicorn/template-indent": "error",
    "unicorn/text-encoding-identifier-case": "off",
    "unicorn/throw-new-error": "error",
    "vue/array-bracket-newline": "off",
    "vue/array-bracket-spacing": "off",
    "vue/arrow-spacing": "off",
    "vue/block-spacing": "off",
    "vue/block-tag-newline": "off",
    "vue/brace-style": "off",
    "vue/comma-dangle": "off",
    "vue/comma-spacing": "off",
    "vue/comma-style": "off",
    "vue/dot-location": "off",
    "vue/func-call-spacing": "off",
    "vue/html-closing-bracket-newline": "off",
    "vue/html-closing-bracket-spacing": "off",
    "vue/html-end-tags": "off",
    "vue/html-indent": "off",
    "vue/html-quotes": "off",
    "vue/html-self-closing": "off",
    "vue/key-spacing": "off",
    "vue/keyword-spacing": "off",
    "vue/max-attributes-per-line": "off",
    "vue/max-len": "off",
    "vue/multiline-html-element-content-newline": "off",
    "vue/mustache-interpolation-spacing": "off",
    "vue/no-extra-parens": "off",
    "vue/no-multi-spaces": "off",
    "vue/no-spaces-around-equal-signs-in-attribute": "off",
    "vue/object-curly-newline": "off",
    "vue/object-curly-spacing": "off",
    "vue/object-property-newline": "off",
    "vue/operator-linebreak": "off",
    "vue/quote-props": "off",
    "vue/script-indent": "off",
    "vue/singleline-html-element-content-newline": "off",
    "vue/space-in-parens": "off",
    "vue/space-infix-ops": "off",
    "vue/space-unary-ops": "off",
    "vue/template-curly-spacing": "off"
  },
  "overrides": [
    {
      "files": [
        "*.ts",
        "*.tsx",
        "*.mts",
        "*.cts"
      ],
      "plugins": [
        "@typescript-eslint"
      ],
      "parser": "@typescript-eslint/parser",
      "parserOptions": {
        "ecmaFeatures": {
          "jsx": true
        },
        "jsxPragma": null,
        "ecmaVersion": "latest",
        "sourceType": "module"
      },
      "rules": {
        "constructor-super": "off",
        "getter-return": "off",
        "no-const-assign": "off",
        "no-dupe-args": "off",
        "no-dupe-class-members": "off",
        "no-dupe-keys": "off",
        "no-func-assign": "off",
        "no-import-assign": "off",
        "no-new-symbol": "off",
        "no-obj-calls": "off",
        "no-redeclare": "off",
        "no-setter-return": "off",
        "no-this-before-super": "off",
        "no-undef": "off",
        "no-unreachable": "off",
        "no-unsafe-negation": "off",
        "no-var": "error",
        "prefer-const": "error",
        "prefer-rest-params": "error",
        "prefer-spread": "error",
        "valid-typeof": "off",
        "@typescript-eslint/adjacent-overload-signatures": "error",
        "@typescript-eslint/ban-ts-comment": "error",
        "@typescript-eslint/ban-types": "error",
        "@typescript-eslint/brace-style": "off",
        "@typescript-eslint/comma-dangle": "off",
        "@typescript-eslint/comma-spacing": "off",
        "@typescript-eslint/func-call-spacing": "off",
        "@typescript-eslint/indent": "off",
        "@typescript-eslint/keyword-spacing": "off",
        "@typescript-eslint/member-delimiter-style": "off",
        "@typescript-eslint/no-array-constructor": "error",
        "@typescript-eslint/no-empty-function": "error",
        "@typescript-eslint/no-empty-interface": "error",
        "@typescript-eslint/no-explicit-any": "warn",
        "@typescript-eslint/no-extra-non-null-assertion": "error",
        "@typescript-eslint/no-extra-parens": "off",
        "@typescript-eslint/no-extra-semi": "off",
        "@typescript-eslint/no-inferrable-types": "error",
        "@typescript-eslint/no-loss-of-precision": "error",
        "@typescript-eslint/no-misused-new": "error",
        "@typescript-eslint/no-namespace": "error",
        "@typescript-eslint/no-non-null-asserted-optional-chain": "error",
        "@typescript-eslint/no-non-null-assertion": "warn",
        "@typescript-eslint/no-this-alias": "error",
        "@typescript-eslint/no-unnecessary-type-constraint": "error",
        "@typescript-eslint/no-unused-vars": "warn",
        "@typescript-eslint/no-var-requires": "error",
        "@typescript-eslint/object-curly-spacing": "off",
        "@typescript-eslint/prefer-as-const": "error",
        "@typescript-eslint/prefer-namespace-keyword": "error",
        "@typescript-eslint/quotes": "off",
        "@typescript-eslint/semi": "off",
        "@typescript-eslint/space-before-blocks": "off",
        "@typescript-eslint/space-before-function-paren": "off",
        "@typescript-eslint/space-infix-ops": "off",
        "@typescript-eslint/triple-slash-reference": "error",
        "@typescript-eslint/type-annotation-spacing": "off"
      }
    }
  ]
}
