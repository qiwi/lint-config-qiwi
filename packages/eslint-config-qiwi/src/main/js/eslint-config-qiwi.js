module.exports = {
  env: {
    node: true,
    browser: true,
    es6: true,
    jest: true,
  },
  parser: '@typescript-eslint/parser',
  extends: [
    'eslint:recommended',
    'standard',
    'plugin:@typescript-eslint/eslint-recommended',
    'plugin:@typescript-eslint/recommended',
    'plugin:react/recommended',
    'plugin:sonarjs/recommended',
    'plugin:array-func/recommended',
  ],
  plugins: [
    'react',
    '@typescript-eslint',
    '@typescript-eslint/eslint-plugin',
    'react-hooks',
    'optimize-regex',
    'sonarjs',
    'unicorn',
  ],
  rules: {
    'eslint/no-unused-vars': 0,
    'optimize-regex/optimize-regex': 'warn',
    'react/prop-types': 0,
    'react/display-name': 0,
    'unicorn/consistent-function-scoping': 'error',
    'unicorn/custom-error-definition': 'off',
    'unicorn/error-message': 'error',
    'unicorn/escape-case': 'error',
    'unicorn/explicit-length-check': 'error',
    'unicorn/import-index': 'error',
    'unicorn/new-for-builtins': 'error',
    'unicorn/no-array-instanceof': 'error',
    'unicorn/no-console-spaces': 'error',
    'unicorn/no-fn-reference-in-iterator': 'off',
    'unicorn/no-for-loop': 'error',
    'unicorn/no-hex-escape': 'error',
    'unicorn/no-keyword-prefix': 'off',
    'no-nested-ternary': 'off',
    'unicorn/no-new-buffer': 'error',
    'unicorn/no-process-exit': 'error',
    'unicorn/no-unreadable-array-destructuring': 'error',
    'unicorn/no-unsafe-regex': 'off',
    'unicorn/no-unused-properties': 'off',
    'unicorn/no-zero-fractions': 'error',
    'unicorn/number-literal-case': 'error',
    'unicorn/prefer-add-event-listener': 'error',
    'unicorn/prefer-dataset': 'error',
    'unicorn/prefer-event-key': 'error',
    'unicorn/prefer-exponentiation-operator': 'error',
    'unicorn/prefer-flat-map': 'error',
    'unicorn/prefer-includes': 'error',
    'unicorn/prefer-node-append': 'error',
    'unicorn/prefer-node-remove': 'error',
    'unicorn/prefer-query-selector': 'error',
    'unicorn/prefer-reflect-apply': 'error',
    'unicorn/prefer-spread': 'error',
    'unicorn/prefer-starts-ends-with': 'error',
    'unicorn/prefer-text-content': 'error',
    'unicorn/prefer-type-error': 'error',
    'unicorn/regex-shorthand': 'error',
    'unicorn/throw-new-error': 'error',
    'array-func/prefer-array-from': 0,
    'sonarjs/cognitive-complexity': 0,
    '@typescript-eslint/no-unused-vars': 'error',
    '@typescript-eslint/interface-name-prefix': 'off',
    'no-useless-constructor': 'off',
    '@typescript-eslint/camelcase': 'off',
    '@typescript-eslint/no-explicit-any': 'off',
    '@typescript-eslint/explicit-function-return-type': 'off',
    '@typescript-eslint/no-use-before-define': 'off',
    '@typescript-eslint/member-delimiter-style': 'off',
    '@typescript-eslint/ban-ts-ignore': 'off',
    'no-return-await': 'error',
    '@typescript-eslint/ban-ts-comment': 0,
    'object-curly-spacing': [2, 'never'],
    'space-before-function-paren': ['error', {
      anonymous: 'never',
      named: 'never',
      asyncArrow: 'always',
    }],
    'comma-dangle': [
      'error', {
        arrays: 'always-multiline',
        exports: 'always-multiline',
        functions: 'never',
        imports: 'always-multiline',
        objects: 'always-multiline',
      },
    ],
  },
  globals: {
    Cypress: 'readonly',
    cy: 'readonly',
    it: 'readonly',
    expect: 'readonly',
    describe: 'readonly',
    fetch: 'readonly',
  },
  settings: {
    react: {
      version: 'latest',
    },
  },
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
    ecmaVersion: 9,
    sourceType: 'module',
  },
}
