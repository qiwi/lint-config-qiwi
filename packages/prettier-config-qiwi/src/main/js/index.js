module.exports = {
  trailingComma: 'all',
  semi: false,
  singleQuote: true,
  importOrder: ['^node:', '^react', '^@qiwi/', '^@pijma/', '^[^./]', '^[./]'],
  importOrderSeparation: true,
  importOrderSortSpecifiers: true,
  importOrderCaseInsensitive: true,
  importOrderGroupNamespaceSpecifiers: true,
  importOrderParserPlugins: ['typescript', 'jsx', 'decorators-legacy', 'importAssertions'],
}
