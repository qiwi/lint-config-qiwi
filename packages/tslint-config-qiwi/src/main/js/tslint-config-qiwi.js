const rules = {
  'brace-style': [
    true,
    'stroustrup',
  ],
  'jsx-boolean-value': false,
  'jsx-no-lambda': false,
  'jsx-no-multiline-js': false,
  'no-unnecessary-type-assertion': false,
  'no-consecutive-blank-lines': [
    true,
    1,
  ],
  'object-curly-spacing': [
    true,
    'never',
  ],
  'object-literal-shorthand': true,
  'one-line': [
    false,
  ],
  'ter-padded-blocks': [
    true,
    {
      'classes': 'always',
    },
  ],
  'prefer-object-spread': true,
  'quotemark': [
    true,
    'single',
    'jsx-double',
  ],
  'space-before-function-paren': [
    true,
    {
      'anonymous': 'never',
      'asyncArrow': 'never',
      'constructor': 'never',
      'method': 'never',
      'named': 'never',
    },
  ],
  'trailing-comma': [
    true,
    {
      'multiline': {
        'arrays': 'always',
        'objects': 'always',
        'functions': 'always',
        'imports': 'always',
        'exports': 'always',
        'typeLiterals': 'never',
      },
      'singleline': 'never',
      'esSpecCompliant': true,
    },
  ],
}

module.exports = {
  extends: [
    'tslint-config-standard',
    'tslint-react',
  ],
  jsRules: rules,
  rules,
}
