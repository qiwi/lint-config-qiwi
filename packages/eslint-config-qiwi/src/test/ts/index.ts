import { resolve } from 'node:path'

import { generateSnapshot } from '@qiwi/stdstream-snapshot'

import * as config from '../../main/js/index.js'

describe('eslint-config-qiwi (TypeScript)', () => {
  it('config is exposed as js', () => {
    expect(config).toEqual(expect.any(Object))
  })

  it('config works as expected', async () => {
    const configPath = resolve(__dirname, '../../main/js/index.js')
    const rulesPath = resolve(__dirname, 'rules/*.ts')
    const eslint = resolve(__dirname, '../../../../../node_modules/.bin/eslint')
    const result = await generateSnapshot({
      cmd: `NODE_OPTIONS=--trace-deprecation ${eslint} --config ${configPath} ${rulesPath} --no-ignore`,
    })

    expect(result).toMatchSnapshot()
  })
})
