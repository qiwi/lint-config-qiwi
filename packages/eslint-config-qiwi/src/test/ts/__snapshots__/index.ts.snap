// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eslint-config-qiwi (TypeScript) config works as expected 1`] = `
Object {
  "err": Object {
    "code": 1,
    "killed": false,
    "signal": null,
  },
  "stderr": "",
  "stdout": "/packages/eslint-config-qiwi/src/test/ts/rules/imports-fail.ts
  1:1  error  Run autofix to sort these imports!  simple-import-sort/imports

/packages/eslint-config-qiwi/src/test/ts/rules/some-rules.ts
   1:20  warning  Missing return type on function                                                                         @typescript-eslint/explicit-module-boundary-types
   1:21  warning  Argument 'foo' should be typed                                                                          @typescript-eslint/explicit-module-boundary-types
   3:7   error    'name' is assigned a value but never used                                                               @typescript-eslint/no-unused-vars
   4:7   error    It's not necessary to initialize 'namefoo' to undefined                                                 no-undef-init
   4:7   error    'namefoo' is assigned a value but never used                                                            @typescript-eslint/no-unused-vars
   4:7   error    'namefoo' is never reassigned. Use 'const' instead                                                      prefer-const
   4:17  error    Do not use useless \`undefined\`                                                                          unicorn/no-useless-undefined
   5:7   error    'message' is assigned a value but never used                                                            @typescript-eslint/no-unused-vars
   5:7   error    'message' is never reassigned. Use 'const' instead                                                      prefer-const
   5:22  error    Unnecessary escape character: \\\\o                                                                        no-useless-escape
   6:3   error    Unexpected var, use let or const instead                                                                no-var
   6:7   error    'pattern1' is assigned a value but never used                                                           @typescript-eslint/no-unused-vars
   6:18  error    Use Unicode escapes instead of hexadecimal escapes                                                      unicorn/no-hex-escape
   6:18  error    Unexpected control character(s) in regular expression: \\\\x1f                                             no-control-regex
   6:18  error    Use uppercase characters for the value of the escape sequence                                           unicorn/escape-case
   6:24  error    Extra semicolon                                                                                         semi
   7:3   error    Unexpected var, use let or const instead                                                                no-var
   7:7   error    'pattern2' is assigned a value but never used                                                           @typescript-eslint/no-unused-vars
   7:18  error    Use a regular expression literal instead of the 'RegExp' constructor                                    prefer-regex-literals
   7:29  error    Use uppercase characters for the value of the escape sequence                                           unicorn/escape-case
   7:29  error    Use Unicode escapes instead of hexadecimal escapes                                                      unicorn/no-hex-escape
   7:29  error    Unexpected control character(s) in regular expression: \\\\x1f                                             no-control-regex
   7:29  error    Strings must use singlequote                                                                            quotes
   7:36  error    Extra semicolon                                                                                         semi
   9:9   error    'myRegex' is assigned a value but never used                                                            @typescript-eslint/no-unused-vars
   9:19  error    Empty class                                                                                             no-empty-character-class
  10:3   error    Object prototype is read only, properties should not be added                                           no-extend-native
  11:3   error    Unexpected var, use let or const instead                                                                no-var
  15:5   error    This case's code block is the same as the block for the case on line 13                                 sonarjs/no-duplicated-branches
  15:5   error    Duplicate case label                                                                                    no-duplicate-case
  15:5   error    Expected a 'break' statement before 'case'                                                              no-fallthrough
  17:5   error    This case's code block is the same as the block for the case on line 15                                 sonarjs/no-duplicated-branches
  17:5   error    Expected a 'break' statement before 'case'                                                              no-fallthrough
  21:10  error    Expected '===' and instead saw '=='                                                                     eqeqeq
  25:7   error    Do not use the '===' operator to compare against -0                                                     no-compare-neg-zero
  26:10  error    This number literal will lose precision at runtime                                                      no-loss-of-precision
  26:10  error    Octal literals should not be used                                                                       no-octal
  27:5   error    Move function declaration to function body root                                                         no-inner-declarations
  27:14  error    'declaration' is defined but never used                                                                 @typescript-eslint/no-unused-vars
  27:25  error    Missing space before function parentheses                                                               space-before-function-paren
  27:28  error    Unexpected empty function 'declaration'                                                                 @typescript-eslint/no-empty-function
  30:3   error    Expected space(s) after \\"if\\"                                                                            keyword-spacing
  30:6   error    There should be no space after this paren                                                               space-in-parens
  30:7   error    {\\"message\\":\\"This always evaluates to truthy. Consider refactoring this code.\\",\\"secondaryLocations\\":]}  sonarjs/no-gratuitous-expressions
  30:7   error    Unexpected constant condition                                                                           no-constant-condition
  30:11  error    There should be no space before this paren                                                              space-in-parens
  31:9   error    Expected a conditional expression and instead saw an assignment                                         no-cond-assign
  31:9   error    Unexpected constant condition                                                                           no-constant-condition
  37:11  error    'z' is assigned a value but never used                                                                  @typescript-eslint/no-unused-vars
  42:45  error    Missing space before =>                                                                                 arrow-spacing
  42:48  error    Missing space after =>                                                                                  arrow-spacing
  45:8   warning  Missing return type on function                                                                         @typescript-eslint/explicit-module-boundary-types
  45:26  error    Missing space before function parentheses                                                               space-before-function-paren
  46:10  error    Redundant use of \`await\` on a return value                                                              no-return-await
  46:21  error    Extra semicolon                                                                                         semi
  50:18  error    Unexpected empty constructor                                                                            @typescript-eslint/no-empty-function
  55:16  warning  Argument 'value' should be typed                                                                        @typescript-eslint/explicit-module-boundary-types
  56:17  error    Extra semicolon                                                                                         semi
  60:8   warning  Missing return type on function                                                                         @typescript-eslint/explicit-module-boundary-types
  61:3   error    The update clause in this loop moves the variable in the wrong direction                                for-direction
  61:8   error    Unexpected var, use let or const instead                                                                no-var
  61:32  error    Empty block statement                                                                                   no-empty
  62:3   error    The update clause in this loop moves the variable in the wrong direction                                for-direction
  62:8   error    Unexpected var, use let or const instead                                                                no-var
  62:33  error    Empty block statement                                                                                   no-empty
  65:1   error    Read-only global 'window' should not be modified                                                        no-global-assign

/packages/eslint-config-qiwi/src/test/ts/rules/trailing-comma-fail.ts
  3:7   error    Missing trailing comma           comma-dangle
  6:27  warning  Missing return type on function  @typescript-eslint/explicit-module-boundary-types
  9:17  error    Unexpected trailing comma        comma-dangle

/packages/eslint-config-qiwi/src/test/ts/rules/trailing-comma.ts
  6:27  warning  Missing return type on function  @typescript-eslint/explicit-module-boundary-types

✖ 71 problems (64 errors, 7 warnings)
  26 errors and 0 warnings potentially fixable with the \`--fix\` option.",
}
`;
